src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +16,36 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
-    }
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+            return country;
+        } catch (Exception ex) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Failed to save the country to the database. "
+                    + ex.getMessage());
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }    }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        Session session;
```

Comment:
It's a good practice to initialize the `Session` variable to `null` to ensure it can be closed properly in a `finally` block if needed.